### PR TITLE
Fix stm32g0b USB stall

### PR DESCRIPTION
**Pull request by marckleinebudde**
Under high CAN-TX load the endpoint on the stm32g0b will stall, this PR fixes the issue.

### CODE REVIEW

**Comment by marckleinebudde** [comment]
@lichtfeind has opened a https://github.com/STMicroelectronics/stm32g0xx_hal_driver/issues/2 to fix this issue upstream.

File: libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h [hunk 1]
```diff
@@ -642,7 +642,7 @@ typedef struct
       (wNBlocks)--; \
     } \
     \
-    (pdwReg)|= (uint32_t)((((wNBlocks) << 26U)) | USB_CNTRX_BLSIZE); \
+    (pdwReg) = ((pdwReg) & ~(USB_CNTRX_BLSIZE | USB_CNTRX_NBLK_MSK)) | (uint32_t)((((wNBlocks) << 26U)) | USB_CNTRX_BLSIZE); \
```
Comment (fenugrec) [inline libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h:645]: why do we clear USB_CNTRX_BLSIZE if we're also  setting it  on the same line - am I missing something ?

File: libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h [hunk 2]
```diff
@@ -709,8 +706,8 @@ typedef struct
   * @param  bEpChNum Endpoint Number.
   * @retval Counter value
   */
-#define USB_DRD_GET_CHEP_TX_CNT(USBx, bEpChNum)           (((USB_DRD_PMA_BUFF + (bEpChNum))->TXBD & 0x03FF0000U) >>16U)
-#define USB_DRD_GET_CHEP_RX_CNT(USBx, bEpChNum)           (((USB_DRD_PMA_BUFF + (bEpChNum))->RXBD & 0x03FF0000U) >>16U)
+#define USB_DRD_GET_CHEP_TX_CNT(USBx, bEpChNum)           (((USB_DRD_PMA_BUFF + (bEpChNum))->TXBD & 0x03FF0000U) >> 16U)
+#define USB_DRD_GET_CHEP_RX_CNT(USBx, bEpChNum)           (((USB_DRD_PMA_BUFF + (bEpChNum))->RXBD & 0x03FF0000U) >> 16U)
```
Comment (fenugrec) [inline libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h:710]: I should frown on unrelated whitespace but I see that's from the upstream ST commit that was cherry-picked

Comment (marckleinebudde) [reply libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h:710]: ACK

File: libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h (hunk 1 above)
Comment (marckleinebudde) [reply libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h:645]: Right - I blindly moving the clearing of the bits to the actual write. Will fix.

File: libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h [hunk 3]
```diff
@@ -670,8 +668,7 @@ typedef struct
     { \
       if ((wCount) == 0U) \
       { \
-        (pdwReg) &= (uint32_t)~USB_CNTRX_NBLK_MSK; \
-        (pdwReg) |= USB_CNTRX_BLSIZE; \
+        (pdwReg) = ((pdwReg) & ~(USB_CNTRX_BLSIZE | USB_CNTRX_NBLK_MSK)) | USB_CNTRX_BLSIZE; \
```
[the remark this answers is not in the document]
Comment (marckleinebudde) [reply libs/STM32_HAL/include/stm32g0xx/stm32g0xx_ll_usb.h:671]: same _excuse_, will fix

**Comment by fenugrec** [comment]
Thanks. Sorry for the delay; I was hoping ST would update the library 'soon' as that would have been preferable to having local fixes (see #157 for my dilemma...)

**Comment by marckleinebudde** [comment]
At least it's on their dashboard now: https://github.com/orgs/STMicroelectronics/projects/3#card-87981733

**Pull request merged**